docs/dev/plugins/cloudformation/transformation/supported-types.md

```diff
@@ -0,0 +1,119 @@
+# Transformation by TOSCA Type
+
+Currently, the CloudFormation plugin only supports certain Node- and RelationshipTypes defined in the [TOSCA Simple Profile](http://docs.oasis-open.org/tosca/TOSCA-Simple-Profile-YAML/v1.1/cos01/). The following sections shows which types are supported and how those are mapped to CloudFormation.
+
+## Supported NodeTypes
+
+The following table contains the NodeTypes currently supported by the CloudFormation plugin and how they're mapped to CloudFormation resources.
+
+| TOSCA NodeType | CloudFormation |
+| --- | --- |
+| Apache | Setup through CloudFormation Init on the EC2 host |
+| Compute | EC2 resource and corresponding SecurityGroup |
+| Database | Setup through CloudFormation Init on the EC2 host |
+| Dbms | Setup through CloudFormation Init on the EC2 host |
+| MysqlDatabase | RDS resource with a mysql engine |
+| MysqlDbms | No specific resource is created |
+| WebApplication | Setup through CloudFormation Init on the EC2 host |
+| Nodejs | Setup through CloudFormation Init on the EC2 host |
+| JavaRuntime | No specific resource is created |
+| JavaApplication | Beanstalk Application and Environment |
+
+## NodeType Transformation Details
+
+These transformations take place in the `TransformationModelNodeVisitor`. During the [transform phase](transformation-workflow.md#transform) of the `TransformationLifeCycle`, the `visit()` of the various NodeTypes are executed in order to fill the `CloudFormationModule` with the necessary information to generate the CloudFormation template and the files needed to deploy the CloudFormation stack. The following section offers a detailed explanation of the transformation behaviour for each of these NodeTypes.
+
+### Compute
+
+Each Compute node that was previously marked in the [prepare phase](transformation-workflow.md#prepare) now gets transformed into an EC2 `Instance` resource with a corresponding `SecurityGroup` resource.
+
+#### 1. Security Group
+
+The `SecurityGroup` is created. The security group allows the opening of ports for the corresponding EC2 instance. If present, optional endpoint ports are opened on this security group.
```

Comment:
`SecurityGroup` also allows other security groups to allow access to themselves (see MysqlDatabase).